src/main/java/com/maxdemarzi/shortest/Service.java

```diff
@@ -438,6 +415,96 @@ private void streamShortestPathsUsingHandwrittenBFS(String centerEmail, List<Str
         }
     }
 
+    private static volatile IntIntMap relationships;
+
+    private static final IntIntMap relationshipCosts(ReadOperations readOps) {
+        if (relationships == null) {
+            Map<String,Integer> costs = ImmutableMap.<String, Integer>builder()
+                .put("EqualTo", 1)
+                .put("HasEmail", 1)
+                .put("AuthoredBy", 2)
+                .put("ContainsEmail", 2)
```

Comment:
i think `ContainsEmail` shouldn't be any cheaper than say `Follows` - or at least should be more than half the cost. being mutually contained in the same document isn't as close a relationship as a social network follow - remember it might be some random webpage